src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2019,81 +1986,119 @@ void Creature::setDeathState(DeathState s, bool despawn)
 
 void Creature::Respawn(bool force)
 {
-    //DestroyForNearbyPlayers(); // pussywizard: not needed
-
     if (force)
     {
         if (IsAlive())
+        {
             setDeathState(DeathState::JustDied);
+        }
         else if (getDeathState() != DeathState::Corpse)
+        {
             setDeathState(DeathState::Corpse);
+        }
     }
 
-    RemoveCorpse(false, false);
+    ConditionList conditions = sConditionMgr->GetConditionsForNotGroupedEntry(CONDITION_SOURCE_TYPE_CREATURE_RESPAWN, GetEntry());
```

Comment:
This isn't as much of a mess as the diff makes it seem - I moved the logic from `Creature::Update()` into `Creature::Respawn()`. Comparing side-by-side is probably easier.